mewarpx/mewarpx/mwxrun.py

```diff
@@ -54,6 +55,22 @@ def init_grid(self, xmin, xmax, zmin, zmax, nx, nz, max_grid_size=4):
         self._set_geom_str()
         self._set_grid_params()
 
+    def init_diags(self):
+        if len(self.simulation.diagnostics) > 0:
+            diag_periods = []
+            for diag in self.simulation.diagnostics:
+                diag_periods.append(diag.period)
+
+            self.step_interval = max(diag_periods)
+            min_period = min(diag_periods)
+            for period in diag_periods:
+                if period % min_period != 0:
```

Comment:
I don't think this is the correct check, you actually want `self.step_interval % period != 0`. Consider eg periods of 10, 100, and 110.